### PR TITLE
feat: add entry point to request node to get monitoring information

### PR DESCRIPTION
**Pull request by vrolland**
## Description of the changes

Add entry point "/status" to get information from the node.

### CODE REVIEW

File: packages/ethereum-storage/src/ethereum-metadata-cache.ts [hunk 1]
```diff
@@ -63,11 +71,43 @@ export default class EthereumMetadataCache {
...
 
     return metadata;
   }
+
+  /**
+   * Get the list of data ids stored
+   *
+   * @returns the list of data ids stored
+   */
+  public async getListDataIds(): Promise<string[]> {
```
Comment (rittme) [inline packages/ethereum-storage/src/ethereum-metadata-cache.ts:89]: same about `List` in the name

File: packages/ethereum-storage/src/ethereum-metadata-cache.ts [hunk 2]
```diff
@@ -63,11 +71,43 @@ export default class EthereumMetadataCache {
...
+    return listDataIds;
+  }
+
+  /**
+   * Update the list of data ids stored
+   *
+   * @param dataId data id to add to the list
+   * @returns
+   */
+  private async updateListDataId(dataId: string): Promise<void> {
```
Comment (rittme) [inline packages/ethereum-storage/src/ethereum-metadata-cache.ts:103]: same about `List` in the name

File: packages/request-client.js/src/http-data-access.ts [hunk 3]
```diff
@@ -194,4 +194,27 @@ export default class HttpDataAccess implements DataAccessTypes.IDataAccess {
 
     return data;
   }
+
+  /**
+   * Gets information
```
Comment (rittme) [inline packages/request-client.js/src/http-data-access.ts:199]: would be good to be more precise. Anything we get in code is information :)

File: packages/request-node/src/requestNode.ts [hunk 4]
```diff
@@ -154,6 +155,20 @@ class RequestNode {
     router.get('/readyz', (_, serverResponse: any) => {
       if (this.initialized) {
         return serverResponse.status(httpStatus.OK).send('OK');
+      }
+      return serverResponse.status(httpStatus.SERVICE_UNAVAILABLE).send(NOT_INITIALIZED_MESSAGE);
+    });
+
+    // Route for information check
+    router.get('/information', (clientRequest: any, serverResponse: any) => {
```
Comment (rittme) [inline packages/request-node/src/requestNode.ts:163]: information feels a bit too generic as a name...

File: packages/data-access/src/ignored-location.ts [hunk 5]
```diff
@@ -0,0 +1,98 @@
+import * as Keyv from 'keyv';
+
+/**
+ * Class used to store the block's reason indexed by location
```
[the remark this answers is not in the document]
Comment (vrolland) [reply packages/data-access/src/ignored-location.ts:4]: `location` is a term we already use in this package. I will make it more understandable tho.

File: packages/request-node/src/requestNode.ts [hunk 6]
```diff
@@ -10,13 +10,14 @@ import Utils from '@requestnetwork/utils';
...
 import getChannelsByTopic from './request/getChannelsByTopic';
+import getStatus from './request/getStatus';
 import getTransactionsByChannelId from './request/getTransactionsByChannelId';
 import ipfsAdd from './request/ipfsAdd';
 import PersistTransaction from './request/persistTransaction';
 import { getEthereumStorage } from './storageUtils';
 
 const NOT_FOUND_MESSAGE =
-  'Not found\nAvailable endpoints:\n/POST persistTransaction\n/GET getTransactionsByChannelId\n/GET getChannelsByTopic\n/POST /ipfsAdd\nGET getConfirmedTransaction';
+  'Not found\nAvailable endpoints:\n/POST persistTransaction\n/GET getTransactionsByChannelId\n/GET getChannelsByTopic\n/POST /ipfsAdd\nGET getConfirmedTransaction\nGET status';
```
Comment (benjlevesque) [inline packages/request-node/src/requestNode.ts:20]:
```suggestion
  'Not found\nAvailable endpoints:\nPOST /persistTransaction\nGET /getTransactionsByChannelId\nGET /getChannelsByTopic\nPOST /ipfsAdd\nGET /getConfirmedTransaction\nGET /status';
```

File: packages/request-client.js/src/http-data-access.ts [hunk 7]
```diff
@@ -194,4 +194,27 @@ export default class HttpDataAccess implements DataAccessTypes.IDataAccess {
...
+  /**
+   * Gets information from the node (version, files etc...)
+   *
+   * @param detailed if true get the list of files hashes
+   */
+  public async _getStatus(detailed?: boolean): Promise<any> {
+    const { data } = await Utils.retry(
+      async () =>
+        axios.get(
+          '/information',
```
Comment (benjlevesque) [inline packages/request-client.js/src/http-data-access.ts:207]: didn't you rename this to /status?

File: packages/ethereum-storage/src/smart-contract-manager.ts [hunk 8]
```diff
@@ -432,6 +432,24 @@ export default class SmartContractManager {
...
 
+  /**
+   * Gets current configuration
+   *
+   * @return the current configuration attributes
+   */
+  public getConfig(): any {
+    return {
+      creationBlockNumberHashStorage: this.creationBlockNumberHashStorage,
+      currentProvider: this.eth.currentProvider.host,
```
Comment (benjlevesque) [inline packages/ethereum-storage/src/smart-contract-manager.ts:443]: have you tested if this is infura, the token wouldn't be included?

**Pull request merged**